pkg/controller/utils/pod/pod_test.go

```diff
@@ -18,7 +18,354 @@ import (
+				},
+			},
+		},
+	}
+	cannotStart, reason = CannotStart(pod)
+	assert.True(t, cannotStart)
+	assert.Equal(t, datadoghqv1alpha1.ExtendedDaemonSetStatusReasonErrImagePull, reason)
+}
+
+func TestPendingCreate(t *testing.T) {
+	now := metav1.Now()
+	pod := newPod(now, true, 5)
+	isPendingCreate := PendingCreate(pod)
+	assert.False(t, isPendingCreate)
+
+	pod.Status.ContainerStatuses = []v1.ContainerStatus{
+		{
+			State: v1.ContainerState{
+				Waiting: &v1.ContainerStateWaiting{
+					Reason: "ContainerCreating",
+				},
+			},
+		},
+	}
+	isPendingCreate = PendingCreate(pod)
+	assert.True(t, isPendingCreate)
+}
+
+func TestHasPodSchedulerIssue(t *testing.T) {
+	pod := ctrltest.NewPod("bar", "pod1", "node1", &ctrltest.NewPodOptions{})
+	hasIssue := HasPodSchedulerIssue(pod)
+	assert.False(t, hasIssue)
+
+	// Has scheduler issue because pod creation time was too long ago
+	pod2 := ctrltest.NewPod("bar", "pod2", "", &ctrltest.NewPodOptions{})
+	hasIssue = HasPodSchedulerIssue(pod2)
+	assert.True(t, hasIssue)
+
+	// Has scheduler issue because pod deletion time was too long ago
+	pod3 := pod.DeepCopy()
+	deletionTs := metav1.NewTime(time.Now().Add(-100 * time.Second))
+	gracePeriod := int64(10)
+	pod3.DeletionTimestamp = &deletionTs
+	pod3.DeletionGracePeriodSeconds = &gracePeriod
+	hasIssue = HasPodSchedulerIssue(pod3)
+	assert.True(t, hasIssue)
+}
+
+func TestUpdatePodCondition(t *testing.T) {
+	status := &v1.PodStatus{
+		Conditions: []v1.PodCondition{
+			{
+				Type:   v1.PodReady,
+				Status: v1.ConditionTrue,
+			},
+		},
+	}
+	condition := &v1.PodCondition{
+		Type:   v1.PodReady,
+		Status: v1.ConditionTrue,
+	}
+	changed := UpdatePodCondition(status, condition)
+	// Condition did not change
+	assert.False(t, changed)
+
+	condition2 := &v1.PodCondition{
+		Type:   v1.PodReady,
+		Status: v1.ConditionFalse,
+	}
+	changed = UpdatePodCondition(status, condition2)
+	// Condition changed
+	assert.True(t, changed)
+}
+
+func TestIsEvicted(t *testing.T) {
+	status := &v1.PodStatus{
+		Conditions: []v1.PodCondition{
+			{
+				Type:   v1.PodReady,
+				Status: v1.ConditionTrue,
+			},
+		},
+	}
+	isEvicted := IsEvicted(status)
+	assert.False(t, isEvicted)
+
+	status2 := &v1.PodStatus{
+		Phase:  v1.PodFailed,
+		Reason: "Evicted",
+	}
+	isEvicted = IsEvicted(status2)
+	assert.True(t, isEvicted)
+}
+
+func TestSortPodByCreationTime(t *testing.T) {
+	time1 := time.Now()
+	time2 := time1.Add(10 * time.Second)
+	time3 := time2.Add(20 * time.Second)
+	pod1 := ctrltest.NewPod("bar", "pod1", "node1", &ctrltest.NewPodOptions{
+		CreationTimestamp: metav1.NewTime(time1),
+	})
+
+	pod2 := ctrltest.NewPod("bar", "pod2", "node1", &ctrltest.NewPodOptions{
+		CreationTimestamp: metav1.NewTime(time2),
+	})
+
+	pod3 := ctrltest.NewPod("bar", "pod3", "node1", &ctrltest.NewPodOptions{
+		CreationTimestamp: metav1.NewTime(time3),
+	})
+	pods := []*v1.Pod{
+		pod2,
+		pod3,
+		pod1,
+	}
+	podList := SortPodByCreationTime(pods)
+	assert.Equal(t, pod3, podList[0])
```

Comment:
This could also be done in a single assert. I think it might make it easier understanding the issue if the test fails.
```go
assert.Equal(t, []*v1.Pod{pod3, pod2, pod1}, podList)
```

Reply:
good point, thanks